File: src/types/nfse.ts
export interface CompNfse {
  nfse: NfseDoc;
}

export interface NfseDoc {
  infNfse: InfNfse;
}

export interface InfNfse {
  numero: string;
  codigoVerificacao: string;
  dataEmissao: string;
  nfseSubstituida?: string;
  outrasInformacoes?: string;
  valoresNfse: ValoresNfse;
  prestadorServico: PrestadorServico;
  tomadorServico?: TomadorServico;
  orgaoGerador?: OrgaoGerador;
  declaracaoPrestacaoServico?: DeclaracaoPrestacaoServico;
}

export interface ValoresNfse {
  baseCalculo: string;
  aliquota?: string;
  valorIss?: string;
  valorLiquidoNfse: string;
}

export interface PrestadorServico {
  identificacaoPrestador: IdentificacaoPrestador;
  razaoSocial: string;
  nomeFantasia?: string;
  endereco: EnderecoNfse;
  contato?: Contato;
}

export interface IdentificacaoPrestador {
  cnpj: string;
  inscricaoMunicipal?: string;
}

export interface TomadorServico {
  identificacaoTomador?: IdentificacaoTomador;
  razaoSocial: string;
  endereco?: EnderecoNfse;
  contato?: Contato;
}

export interface IdentificacaoTomador {
  cnpj?: string;
  cpf?: string;
  inscricaoMunicipal?: string;
}

export interface EnderecoNfse {
  endereco?: string;
  numero?: string;
  complemento?: string;
  bairro?: string;
  codigoMunicipio?: string;
  uf?: string;
  cep?: string;
}

export interface Contato {
  telefone?: string;
  email?: string;
}

export interface OrgaoGerador {
  codigoMunicipio: string;
  uf: string;
}

export interface DeclaracaoPrestacaoServico {
  infDeclaracaoPrestacaoServico: InfDeclaracaoPrestacaoServico;
}

export interface InfDeclaracaoPrestacaoServico {
  competencia?: string;
  servico: Servico;
  prestador: { cnpj: string; inscricaoMunicipal?: string };
  tomador?: TomadorServico;
  optanteSimplesNacional?: string;
  incentivoFiscal?: string;
}

export interface Servico {
  valores: ValoresServico;
  itemListaServico: string;
  codigoCnae?: string;
  codigoTributacaoMunicipio?: string;
  discriminacao: string;
  codigoMunicipio: string;
}

export interface ValoresServico {
  valorServicos: string;
  valorDeducoes?: string;
  valorPis?: string;
  valorCofins?: string;
  valorInss?: string;
  valorIr?: string;
  valorCsll?: string;
  issRetido?: string;
  valorIss?: string;
  valorIssRetido?: string;
  outrasRetencoes?: string;
  baseCalculo?: string;
  aliquota?: string;
  valorLiquidoNfse?: string;
  descontoIncondicionado?: string;
  descontoCondicionado?: string;
}

// ============================================================
// NFS-e Padrão Nacional (SPED)
// ============================================================

export interface SpedCompNfse {
  infNFSe: SpedInfNfse;
}

export interface SpedInfNfse {
  id: string;
  versao: string;
  nNFSe: string;
  xLocEmi: string;
  xLocPrestacao: string;
  cLocIncid: string;
  xLocIncid: string;
  xTribNac: string;
  verAplic: string;
  ambGer: string;
  tpEmis: string;
  procEmi: string;
  cStat: string;
  dhProc: string;
  nDFSe: string;
  emit: SpedEmit;
  valores: SpedValoresNfse;
  dps?: SpedDps;
}

export interface SpedEmit {
  CNPJ: string;
  xNome: string;
  enderNac: SpedEndereco;
  fone?: string;
  email?: string;
}

export interface SpedEndereco {
  xLgr: string;
  nro: string;
  xCpl?: string;
  xBairro: string;
  cMun: string;
  UF: string;
  CEP: string;
}

export interface SpedValoresNfse {
  vLiq: string;
}

export interface SpedDps {
  infDPS: SpedInfDps;
}

export interface SpedInfDps {
  id: string;
  tpAmb: string;
  dhEmi: string;
  verAplic: string;
  serie: string;
  nDPS: string;
  dCompet: string;
  tpEmit: string;
  cLocEmi: string;
  prest: SpedPrestador;
  toma: SpedTomador;
  serv: SpedServico;
  valores: SpedValoresDps;
}

export interface SpedPrestador {
  CNPJ: string;
  xNome?: string;
  fone?: string;
  email?: string;
  regTrib?: SpedRegTrib;
}

export interface SpedRegTrib {
  opSimpNac: string;
  regApTribSN: string;
  regEspTrib: string;
}

export interface SpedTomador {
  CNPJ?: string;
  CPF?: string;
  xNome: string;
  end?: SpedTomadorEnd;
  email?: string;
}

export interface SpedTomadorEnd {
  endNac: SpedEnderecoNac;
  xLgr: string;
  nro: string;
  xCpl?: string;
  xBairro: string;
}

export interface SpedEnderecoNac {
  cMun: string;
  CEP: string;
}

export interface SpedServico {
  locPrest: SpedLocPrest;
  cServ: SpedCServ;
}

export interface SpedLocPrest {
  cLocPrestacao: string;
}

export interface SpedCServ {
  cTribNac: string;
  xDescServ: string;
}

export interface SpedValoresDps {
  vServPrest: SpedVServPrest;
  trib?: SpedTributos;
}

export interface SpedVServPrest {
  vServ: string;
}

export interface SpedTributos {
  tribMun: SpedTribMun;
  tribFed?: SpedTribFed;
  totTrib?: SpedTotTrib;
}

export interface SpedTribMun {
  tribISSQN: string;
  tpRetISSQN: string;
}

export interface SpedTribFed {
  piscofins: SpedPisCofins;
}

export interface SpedPisCofins {
  CST: string;
}

export interface SpedTotTrib {
  pTotTribSN: string;
}

File: src-tauri/src/models/nfse.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CompNfse {
    pub nfse: NfseDoc,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NfseDoc {
    pub inf_nfse: InfNfse,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct InfNfse {
    pub numero: String,
    pub codigo_verificacao: String,
    pub data_emissao: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub nfse_substituida: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub outras_informacoes: Option<String>,
    pub valores_nfse: ValoresNfse,
    pub prestador_servico: PrestadorServico,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub tomador_servico: Option<TomadorServico>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub orgao_gerador: Option<OrgaoGerador>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub declaracao_prestacao_servico: Option<DeclaracaoPrestacaoServico>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ValoresNfse {
    pub base_calculo: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub aliquota: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub valor_iss: Option<String>,
    pub valor_liquido_nfse: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PrestadorServico {
    pub identificacao_prestador: IdentificacaoPrestador,
    pub razao_social: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub nome_fantasia: Option<String>,
    pub endereco: EnderecoNfse,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub contato: Option<Contato>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct IdentificacaoPrestador {
    pub cnpj: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub inscricao_municipal: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct TomadorServico {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub identificacao_tomador: Option<IdentificacaoTomador>,
    pub razao_social: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub endereco: Option<EnderecoNfse>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub contato: Option<Contato>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct IdentificacaoTomador {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub cnpj: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub cpf: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub inscricao_municipal: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct EnderecoNfse {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub endereco: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub numero: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub complemento: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub bairro: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub codigo_municipio: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub uf: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub cep: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Contato {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub telefone: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub email: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct OrgaoGerador {
    pub codigo_municipio: String,
    pub uf: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DeclaracaoPrestacaoServico {
    pub inf_declaracao_prestacao_servico: InfDeclaracaoPrestacaoServico,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct InfDeclaracaoPrestacaoServico {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub competencia: Option<String>,
    pub servico: Servico,
    pub prestador: PrestadorRef,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub tomador: Option<TomadorServico>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub optante_simples_nacional: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub incentivo_fiscal: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PrestadorRef {
    pub cnpj: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub inscricao_municipal: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Servico {
    pub valores: ValoresServico,
    pub item_lista_servico: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub codigo_cnae: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub codigo_tributacao_municipio: Option<String>,
    pub discriminacao: String,
    pub codigo_municipio: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ValoresServico {
    pub valor_servicos: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub valor_deducoes: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub valor_pis: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub valor_cofins: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub valor_inss: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub valor_ir: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub valor_csll: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub iss_retido: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub valor_iss: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub valor_iss_retido: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub outras_retencoes: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub base_calculo: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub aliquota: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub valor_liquido_nfse: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub desconto_incondicionado: Option<String>,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub desconto_condicionado: Option<String>,
}
